### PR TITLE
(GH-33) Add Note about Chocolatey Dependency

### LINKED ISSUE #33: Note: Chocolatey Dependency
Unless a package needs a specific version of Chocolatey in order to function, a dependency on the Chocolatey Package is not required.

Add this as a note so that a human can verify.

### PR DESCRIPTION
**Pull request by gep13**
Closes #33

### CODE REVIEW

File: src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace chocolatey.package.validator.infrastructure.app.rules
+{
+    using System.Linq;
+    using infrastructure.rules;
+    using NuGet;
+
+    public class IncludesChocolateyDependencyNote : BasePackageRule
+    {
+        public override string ValidationFailureMessage { get { return "Package takes a dependency on Chocolatey, which should only be required if package uses a specific Chocolatey feature.  Please see [here](https://github.com/chocolatey/package-validator/wiki/ChocolateyDependency) for further information and guidance."; } }
```
Comment (ferventcoder) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: Here

File: src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        protected override PackageValidationOutput is_valid(IPackage package)
+        {
+            return package.DependencySets.All(dependencySet => dependencySet.Dependencies.All(dependency => dependency.Id.ToLower() != "chocolatey"));
```
Comment (ferventcoder) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: Is `.All` a LINQ thing?

use .to_lower() - null safe

Comment (ferventcoder) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: I would have thought it was a `.Any()`?

Comment (gep13) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: Yes.

Sent from my Windows Phone
________________________________

Comment (gep13) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: I did what ReSharper told me to do :-)

I started with a nested foreach loop, and ReSharper refactored the logic :-)

Sent from my Windows Phone
________________________________

Comment (gep13) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: @ferventcoder are you happy to trust ReSharper in this instance?

**Comment by gep13** [comment]
@ferventcoder updated with suggested changes.

File: src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs (hunk 2 above)
Comment (ferventcoder) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: most of the time :)

Comment (ferventcoder) [inline src/chocolatey.package.validator/infrastructure.app/rules/IncludesChocolateyDependencyNote.cs]: Did you get any flagging items with this one?

**Pull request merged**